backend/provision/helpers/tardis.sql

```diff
@@ -0,0 +1,81 @@
+SET statement_timeout = 0;
+SET lock_timeout = 0;
+SET client_encoding = 'UTF8';
+SET standard_conforming_strings = on;
+SET check_function_bodies = false;
+SET client_min_messages = warning;
+SET row_security = off;
+
+
+CREATE SCHEMA history;
+
+CREATE EXTENSION IF NOT EXISTS plpgsql WITH SCHEMA pg_catalog;
+
+CREATE EXTENSION IF NOT EXISTS btree_gist WITH SCHEMA public;
+
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp" WITH SCHEMA public;
+
+SET search_path = history, pg_catalog;
+
+CREATE FUNCTION log_change() RETURNS trigger
+    LANGUAGE plpgsql
+    AS $_$
+    DECLARE
+      c refcursor;
+      tt tstzrange;
+    BEGIN
+        IF TG_OP = 'INSERT' THEN
+            EXECUTE 'INSERT INTO history.' || TG_TABLE_NAME ||
+              ' SELECT $1.*, tstzrange(now(), $$infinity$$, $$[)$$)' USING NEW;
+            RETURN NEW;
+        ELSIF TG_OP = 'UPDATE' THEN
+            OPEN c FOR EXECUTE 'SELECT _validrange FROM history.' || TG_TABLE_NAME ||
+              ' WHERE id = $1 ORDER BY _validrange DESC LIMIT 1 FOR UPDATE'
+              USING NEW.id;
+            FETCH FROM c INTO tt;
+
+            IF isempty(tstzrange(lower(tt), now(), $$[)$$)) THEN
+
+                -- What edge case are we missing by commenting the following lines of code?
```

Comment:
@kardan we should remove this commented code, we know what _edge cases_ are we missing